src/app/organisms/settings/CrossSigning.jsx

```diff
@@ -19,57 +20,61 @@ import SettingTile from '../../molecules/setting-tile/SettingTile';
 import { authRequest } from './AuthRequest';
 import { useCrossSigningStatus } from '../../hooks/useCrossSigningStatus';
 
-const failedDialog = () => {
-  const renderFailure = (requestClose) => (
-    <div className="cross-signing__failure">
-      <Text variant="h1">{twemojify('❌')}</Text>
-      <Text weight="medium">Failed to setup cross signing. Please try again.</Text>
-      <Button onClick={requestClose}>Close</Button>
-    </div>
-  );
+import '../../i18n';
 
-  openReusableDialog(
-    <Text variant="s1" weight="medium">Setup cross signing</Text>,
-    renderFailure,
-  );
-};
+function CrossSigningSetup() {
+  const { t } = useTranslation();
 
-const securityKeyDialog = (key) => {
-  const downloadKey = () => {
-    const blob = new Blob([key.encodedPrivateKey], {
-      type: 'text/plain;charset=us-ascii',
-    });
-    FileSaver.saveAs(blob, 'security-key.txt');
-  };
-  const copyKey = () => {
-    copyToClipboard(key.encodedPrivateKey);
-  };
+  const initialValues = { phrase: '', confirmPhrase: '' };
+  const [genWithPhrase, setGenWithPhrase] = useState(undefined);
 
-  const renderSecurityKey = () => (
-    <div className="cross-signing__key">
-      <Text weight="medium">Please save this security key somewhere safe.</Text>
-      <Text className="cross-signing__key-text">
-        {key.encodedPrivateKey}
-      </Text>
-      <div className="cross-signing__key-btn">
-        <Button variant="primary" onClick={() => copyKey(key)}>Copy</Button>
-        <Button onClick={() => downloadKey(key)}>Download</Button>
+  const failedDialog = () => {
+    const renderFailure = (requestClose) => (
+      <div className="cross-signing__failure">
+        <Text variant="h1">{twemojify('❌')}</Text>
+        <Text weight="medium">{t('Organisms.CrossSigning.setup_failed')}</Text>
+        <Button onClick={requestClose}>{t('common.close')}</Button>
       </div>
-    </div>
-  );
+    );
 
-  // Download automatically.
-  downloadKey();
+    openReusableDialog(
+      <Text variant="s1" weight="medium">{t('Organisms.CrossSigning.setup')}</Text>,
+      renderFailure,
+    );
+  };
 
-  openReusableDialog(
-    <Text variant="s1" weight="medium">Security Key</Text>,
-    () => renderSecurityKey(),
-  );
-};
```

Comment:
Can't we do translation without moving these inside component?

Reply:
I think it needs to be inside component for the t() hook to function properly. Is there a particular reason to not have these inside component?

Reply:
you can use `t()`, but `<Trans i18nKey="" />` seems to work